src/TRE-API/DoAgentWork.cs

```diff
@@ -52,581 +54,304 @@ public DoAgentWork(IServiceProvider serviceProvider,
+            if (!response.IsSuccessStatusCode)
+            {
+                Log.Error("TESK status check failed for {taskID}", taskID);
+                return;
+            }
 
-            HttpClientHandler handler = new HttpClientHandler();
+            var content = await response.Content.ReadAsStringAsync();
+            var status = JsonConvert.DeserializeObject<TESKstatus>(content);
+            var existing = _dbContext.TESK_Status.FirstOrDefault(x => x.id == status.id);
 
-            if (_AgentSettings.Proxy)
+            bool shouldReport = existing == null || existing.state != status.state;
+            if (shouldReport)
             {
-                handler = new HttpClientHandler
-                {
-                    Proxy = new WebProxy(_AgentSettings.ProxyAddresURL, true), // Replace with your proxy server URL
-                    UseProxy = _AgentSettings.Proxy,
-                };
+                if (existing == null)
+                    _dbContext.Add(status);
+                else { existing.state = status.state; _dbContext.Update(existing); }
+                _dbContext.SaveChanges();
             }
 
+            if (!shouldReport && !TerminalStates.Contains(status.state)) return;
 
-            using var httpClient = new HttpClient(handler);
-            // Define the URL for the POST request
-            string apiUrl = _AgentSettings.TESKAPIURL;
-
-            // Create a HttpRequestMessage with the HTTP method set to POST
-            var request = new HttpRequestMessage(HttpMethod.Post, apiUrl);
+            await HandleTESKFinalStatus(status.state, subId, taskID);
 
-            // Set the headers
-            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-            request.Headers.TryAddWithoutValidation("Content-Type", "application/json");
+            if (TerminalStates.Contains(status.state))
+                await FinalizeFilesAndNotify(subId, outputBucket, tesId, NameTes);
+        }
 
-            // Attach the JSON string to the request's content
-            request.Content = new StringContent(jsonContent, Encoding.UTF8, "application/json");
+        private async Task HandleTESKFinalStatus(string state, int subId, string taskID)
+        {
+            var token = _dbContext.TokensToExpire.FirstOrDefault(x => x.SubId == subId);
+            if (token != null)
+            {
+                _dbContext.TokensToExpire.Remove(token);
+                _hasuraAuthenticationService.ExpirerToken(token.Token);
+                _dbContext.SaveChanges();
+            }
 
+            var statusType = state switch
+            {
+                "QUEUED" => StatusType.TransferredToPod,
+                "RUNNING" => StatusType.PodProcessing,
+                "COMPLETE" => StatusType.PodProcessingComplete,
+                _ => StatusType.Cancelled
+            };
+
+            _subHelper.UpdateStatusForTre(subId.ToString(), statusType, "");
```

Comment:
I haven't been able to get it to go past the "Completed" stage. It picks up the job, sends it to TES, but then it fails to send it to Egress. Specifically on this line it fails to update the status, so it never gets to the point of sending it to egress